vllm/v1/sample/sampler.py

```diff
@@ -134,3 +137,30 @@ def apply_penalties(
                 sampling_metadata.repetition_penalties,
                 sampling_metadata.output_token_ids)
         return logits
+
+    def apply_logits_processors(
+        self,
+        logits: torch.Tensor,
+        sampling_metadata: SamplingMetadata,
+    ) -> torch.Tensor:
+        for seq_idx, seq_logits_processors in \
+            enumerate(sampling_metadata.logits_processors):
+            if not seq_logits_processors:
+                continue
```

Comment:
It seems like a mandatory performance loss that we will always iterate through this list `max_num_reqs` times, even if there are no logit processors. Would it be worse to use a dynamically allocated Dict or structure that only includes the seqs that have logit processors?

Reply:
I think it's ok to use a dict here